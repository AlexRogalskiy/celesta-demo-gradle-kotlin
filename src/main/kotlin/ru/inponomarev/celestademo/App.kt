package ru.inponomarev.celestademo

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication

@SpringBootApplication
@EnableConfigurationProperties(AppSettings::class)
open class App

fun main(args: Array<String>) {
    //NB: System restart does not clean up H2 database, thus
    //leading to wrong integration test results
    System.setProperty("spring.devtools.restart.enabled", "false")
    runApplication<App>(*args)
}