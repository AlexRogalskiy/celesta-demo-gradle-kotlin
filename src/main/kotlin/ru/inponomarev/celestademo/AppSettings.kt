package ru.inponomarev.celestademo

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "celestademo")
class AppSettings {
    var initdemodata: Boolean? = null
}
