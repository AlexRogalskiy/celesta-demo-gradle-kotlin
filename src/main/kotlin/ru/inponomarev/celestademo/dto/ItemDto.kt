package ru.inponomarev.celestademo.dto

class ItemDto(
    val id: String?,
    val name: String?,
    val orderedQuantity: Int?
)