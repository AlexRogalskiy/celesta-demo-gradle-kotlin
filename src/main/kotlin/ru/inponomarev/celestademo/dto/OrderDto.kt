package ru.inponomarev.celestademo.dto

class OrderDto(
    val id: Int? = null,
    val customerId: Int? = null,
    val itemId: String? = null,
    val quantity: Int? = null,
    val price: Int? = null,
    val amount: Int? = null
)