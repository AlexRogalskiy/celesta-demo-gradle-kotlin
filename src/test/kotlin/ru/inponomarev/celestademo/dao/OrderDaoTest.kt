package ru.inponomarev.celestademo.dao

import ru.curs.celestaunit.CelestaTest
import ru.inponomarev.celestademo.CustomerCursor
import ru.inponomarev.celestademo.ItemCursor
import org.junit.jupiter.api.BeforeEach
import ru.curs.celesta.CallContext
import ru.inponomarev.celestademo.dto.OrderDto
import ru.inponomarev.celestademo.OrderCursor
import org.approvaltests.Approvals
import com.fasterxml.jackson.databind.ObjectMapper
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

@CelestaTest
class OrderDaoTest {
    private val orderDao = OrderDao()
    private lateinit var customer: CustomerCursor
    private lateinit var item: ItemCursor

    @BeforeEach
    fun setUp(ctx: CallContext?) {
        customer = CustomerCursor(ctx)
        customer.setName("John Doe")
            .setEmail("john@example.com")
            .insert()
        item = ItemCursor(ctx)
        item.setId("12345")
            .setName("cheese")
            .setDefaultPrice(42)
            .insert()
    }

    @Test
    fun orderIsPostedWithDefaultPrice(ctx: CallContext?) {
        //ARRANGE
        val dto = OrderDto(
            customerId = customer.id,
            itemId = item.id,
            quantity = 100
        )

        //ACT
        val result = orderDao.postOrder(ctx, dto)

        //ASSERT
        val orderCursor = OrderCursor(ctx)
        Assertions.assertThat(orderCursor.count()).isEqualTo(1)
        orderCursor.first()
        Assertions.assertThat(orderCursor.price).isEqualTo(item.defaultPrice)
        Assertions.assertThat(orderCursor.quantity).isEqualTo(100)
        Assertions.assertThat(orderCursor.amount).isEqualTo(item.defaultPrice * orderCursor.quantity)
        Approvals.verifyJson(ObjectMapper().writer().writeValueAsString(result))
    }

    @Test
    fun orderedItemsMethodReturnsAggregatedValues(ctx: CallContext?) {
        //ARRANGE
        val item2 = ItemCursor(ctx)
        item2.setId("2").setName("item 2").insert()
        val orderCursor = OrderCursor(ctx)
        orderCursor.setId(null).setItemId(item.id).setCustomerId(customer.id).setQuantity(1).insert()
        orderCursor.setId(null).setItemId(item.id).setCustomerId(customer.id).setQuantity(3).insert()
        orderCursor.setId(null).setItemId(item2.id).setCustomerId(customer.id).setQuantity(5).insert()
        //ACT
        val result = orderDao.getItems(ctx)

        //ASSERT
        Approvals.verifyJson(ObjectMapper().writer().writeValueAsString(result))
    }
}