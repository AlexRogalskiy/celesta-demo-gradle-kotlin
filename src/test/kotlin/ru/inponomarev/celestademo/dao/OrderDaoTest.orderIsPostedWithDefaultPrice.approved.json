{
  "id": 1,
  "customerId": 1,
  "itemId": "12345",
  "quantity": 100,
  "price": 42,
  "amount": 4200
}